lib/astc_codec.cpp

```diff
@@ -291,38 +293,242 @@ astcVkFormat(ktx_uint32_t block_size, bool sRGB) {
 
 /**
  * @memberof ktxTexture
- * @ingroup writer
+ * @internal
+ * @ingroup reader
  * @~English
- * @brief Creates valid ASTC encoder action from string.
+ * @brief       Should be used to get uncompressed version of ASTC VkFormat
  *
- * @return      Valid astc_profile from string
+ * The decompressed format is calculated from coressponding ASTC format. There are
+ * only 3 possible options currently supported. RGBA8, sRGBA8 and RGBA32.
+ *
+ * @return      Uncompressed version of VKFormat for a specific ASTC VkFormat
  */
-static astcenc_profile
-astcEncoderAction(const ktxAstcParams &params, const uint32_t* bdb) {
+inline VkFormat getUncompressedFormat(VkFormat format) noexcept {
```

Comment:
You can use the DFD for this. Pass the ktxTexture2 here instead of the VkFormat. You've already checked the colorModel in `ktxTexture2_DecodeAstc` so you can do

```
inline VkFormat getUncompressedFormat(ktxTexture2* This) noexcept {
    if (DFDSVAL(This->pDfd + 1, 0, QUALIFIERS) & KHR_DF_SAMPLE_DATATYPE_FLOAT) {
        return VK_FORMAT_R32G32B32A32_SFLOAT;
    } else {
        if (This->GetTransfer_e() == KHR_DF_TRANSFER_SRGB)
            return VK_FORMAT_R8G8B8A8_SRGB;
        else
            return VK_FORMAT_R8G8B8A8_UNORM;
    }
}
```

Reply:
Thank you. fixed in https://github.com/KhronosGroup/KTX-Software/pull/967/commits/b7acc0aa7d4b45c8286a341dae73bdd8101fe21c